$OpenBSD: patch-groupsock_GroupsockHelper_cpp,v 1.1 2012/02/28 17:55:47 stsp Exp $
Fix "Unable to determine our source address: This computer has an
invalid IP address: 0x7f000001"
--- groupsock/GroupsockHelper.cpp.orig	Sat Feb  4 12:01:47 2012
+++ groupsock/GroupsockHelper.cpp	Sun Feb 26 14:51:00 2012
@@ -582,7 +582,7 @@ netAddressBits ourIPAddress(UsageEnvironment& env) {
 	break;
       }
 
-      loopbackWorks = 1;
+      loopbackWorks = !badAddressForUs(fromAddr.sin_addr.s_addr);
     } while (0);
 
     if (sock >= 0) {
